lib/effects/effects.dart

```diff
@@ -43,13 +43,11 @@ abstract class ComponentEffect<T extends BaseComponent> {
     this._initialIsInfinite,
     this._initialIsAlternating, {
     this.isRelative = false,
-    this.curve = Curves.linear,
+    Curve curve,
     this.onComplete,
-  }) {
-    isInfinite = _initialIsInfinite;
-    isAlternating = _initialIsAlternating;
-    curve ??= Curves.linear;
-  }
+  })  : isInfinite = _initialIsInfinite,
+        isAlternating = _initialIsAlternating,
+        curve = curve ?? Curves.linear;
```

Comment:
Is this line needed now when the default is moved up to the named arguments?

Reply:
In the `null-safety` branch `curve` argument os nullable. This effectively means that it could receive `null` and default value won't be applied. Thus, we provide `Curves.linear` as a value to replace the passed `null` value. If the value of `curve` is not passed at all it defaults to `null` and the same code is in effect again.

Reply:
Actually, here is another issue. I would rather add an assert for `isRelative` here as it could be passed as `null` in this before-null-safety code and that disables the default value `false` defined for it.

Reply:
> Actually, here is another issue. I would rather add an assert for `isRelative` here as it could be passed as `null` in this before-null-safety code and that disables the default value `false` defined for it.

Added the assert for `isRelative`